core/Test/function_test.cpp

```diff
@@ -608,7 +615,108 @@ void test_RSA_sign_verify()
         SAFE_FREE(returnJsonChar);
         log_i("============%s end==========\n", plaintext[i].c_str());
     }
-    log_i("============test_RSA_sign_verify end==========\n");
+    log_i("============test_RSA_sign_verify_RAW end==========\n");
+}
+
+/*
+
+step1. generate an rsa 3072 key as the CM(customer master key)
+
+step2. Sign the digest
+
+step3. Verify the signature
+
+*/
+void test_RSA_sign_verify_DIGEST()
+{
+    log_i("============test_RSA_sign_verify_DIGEST start==========\n");
+    std::string plaintext[] = {"Test1234-RSA2048", "Test1234-RSA3072", "Test1234-RSA4096"};
+    uint32_t keyspec[] = {EH_RSA_2048, EH_RSA_3072, EH_RSA_4096};
+
+    case_number += sizeof(plaintext) / sizeof(plaintext[0]);
+    for (int i = 0; i < sizeof(plaintext) / sizeof(plaintext[0]); i++)
+    {
+        log_i("============%s start==========\n", plaintext[i].c_str());
+        ehsm_status_t ret = EH_OK;
+        char *returnJsonChar = (char *)calloc(10000, sizeof(char));
+
+        char *cmk_base64 = nullptr;
+        char *signature_base64 = nullptr;
+        bool result = false;
+        RetJsonObj retJsonObj;
+
+        JsonObj param_json;
+        JsonObj payload_json;
+
+        payload_json.addData_uint32("keyspec", keyspec[i]);
+        payload_json.addData_uint32("origin", EH_INTERNAL_KEY);
+        payload_json.addData_uint32("keyusage", EH_KEYUSAGE_SIGN_VERIFY);
+        param_json.addData_uint32("action", EH_CREATE_KEY);
+        param_json.addData_JsonValue("payload", payload_json.getJson());
+
+        EHSM_FFI_CALL(param_json.toString().c_str(), returnJsonChar);
+        retJsonObj.parse(returnJsonChar);
+        if (retJsonObj.getCode() != 200)
+        {
+            log_e("FFI_CreateKey failed, error message: %s \n", retJsonObj.getMessage().c_str());
+            goto cleanup;
+        }
+        log_i("FFI_CreateKey Json : %s\n", returnJsonChar);
+        log_i("Create CMK with RAS SUCCESSFULLY!\n");
+
+        cmk_base64 = retJsonObj.readData_cstr("cmk");
+
+        payload_json.clear();
+        payload_json.addData_string("cmk", cmk_base64);
+        payload_json.addData_string("message", "JVAPBOYcL7HFfJhtEwqL1lDoMZnUVwxYpCa6atFTH0E=");
```

Comment:
what's does "JVAPBOYcL7HFfJhtEwqL1lDoMZnUVwxYpCa6atFTH0E=" mean?

Reply:
just a sha256 string

Reply:
so " std::string plaintext[] = {"Test1234-RSA2048", "Test1234-RSA3072", "Test1234-RSA4096"};" is not used anymore?

Reply:
is this comment handled?

Reply:
> so " std::string plaintext[] = {"Test1234-RSA2048", "Test1234-RSA3072", "Test1234-RSA4096"};" is not used anymore?

yes, I'll delete it